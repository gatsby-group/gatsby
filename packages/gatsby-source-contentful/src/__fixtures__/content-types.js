export const contentTypes = [
  {
    sys: {
      space: {
        sys: {
          type: `Link`,
          linkType: `Space`,
          id: `k8iqpp6u0ior`,
        },
      },
      id: `number`,
      type: `ContentType`,
      createdAt: `2021-03-01T16:59:23.010Z`,
      updatedAt: `2021-05-21T11:20:58.851Z`,
      environment: {
        sys: {
          id: `master`,
          type: `Link`,
          linkType: `Environment`,
        },
      },
      revision: 5,
    },
    displayField: `title`,
    name: `Number`,
    description: ``,
    fields: [
      {
        id: `title`,
        name: `Title`,
        type: `Symbol`,
        localized: false,
        required: true,
        disabled: false,
        omitted: false,
      },
      {
        id: `integer`,
        name: `Integer`,
        type: `Integer`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `integerLocalized`,
        name: `Integer Localized`,
        type: `Integer`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `decimal`,
        name: `Decimal`,
        type: `Number`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `decimalLocalized`,
        name: `Decimal Localized`,
        type: `Number`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
      },
    ],
  },
  {
    sys: {
      space: {
        sys: {
          type: `Link`,
          linkType: `Space`,
          id: `k8iqpp6u0ior`,
        },
      },
      id: `text`,
      type: `ContentType`,
      createdAt: `2021-03-01T17:02:35.612Z`,
      updatedAt: `2021-05-21T10:48:06.210Z`,
      environment: {
        sys: {
          id: `master`,
          type: `Link`,
          linkType: `Environment`,
        },
      },
      revision: 3,
    },
    displayField: `title`,
    name: `Text`,
    description: ``,
    fields: [
      {
        id: `title`,
        name: `Title`,
        type: `Symbol`,
        localized: false,
        required: true,
        disabled: false,
        omitted: false,
      },
      {
        id: `short`,
        name: `Short`,
        type: `Symbol`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `shortLocalized`,
        name: `Short Localized`,
        type: `Symbol`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `shortList`,
        name: `Short List`,
        type: `Array`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
        items: {
          type: `Symbol`,
          validations: [],
        },
      },
      {
        id: `shortListLocalized`,
        name: `Short List Localized`,
        type: `Array`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
        items: {
          type: `Symbol`,
          validations: [],
        },
      },
      {
        id: `longPlain`,
        name: `Long Plain`,
        type: `Text`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `longMarkdown`,
        name: `Long Markdown`,
        type: `Text`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `longLocalized`,
        name: `Long Localized`,
        type: `Text`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
      },
    ],
  },
  {
    sys: {
      space: {
        sys: {
          type: `Link`,
          linkType: `Space`,
          id: `k8iqpp6u0ior`,
        },
      },
      id: `mediaReference`,
      type: `ContentType`,
      createdAt: `2021-03-01T17:03:21.639Z`,
      updatedAt: `2021-05-21T11:11:20.265Z`,
      environment: {
        sys: {
          id: `master`,
          type: `Link`,
          linkType: `Environment`,
        },
      },
      revision: 8,
    },
    displayField: `title`,
    name: `Media Reference`,
    description: ``,
    fields: [
      {
        id: `title`,
        name: `Title`,
        type: `Symbol`,
        localized: false,
        required: true,
        disabled: false,
        omitted: false,
      },
      {
        id: `one`,
        name: `One`,
        type: `Link`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
        linkType: `Asset`,
      },
      {
        id: `oneLocalized`,
        name: `One Localized`,
        type: `Link`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
        linkType: `Asset`,
      },
      {
        id: `many`,
        name: `Many`,
        type: `Array`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
        items: {
          type: `Link`,
          validations: [],
          linkType: `Asset`,
        },
      },
      {
        id: `manyLocalized`,
        name: `Many Localized`,
        type: `Array`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
        items: {
          type: `Link`,
          validations: [],
          linkType: `Asset`,
        },
      },
    ],
  },
  {
    sys: {
      space: {
        sys: {
          type: `Link`,
          linkType: `Space`,
          id: `k8iqpp6u0ior`,
        },
      },
      id: `boolean`,
      type: `ContentType`,
      createdAt: `2021-03-01T17:05:40.030Z`,
      updatedAt: `2021-05-21T10:32:45.505Z`,
      environment: {
        sys: {
          id: `master`,
          type: `Link`,
          linkType: `Environment`,
        },
      },
      revision: 3,
    },
    displayField: `title`,
    name: `Boolean`,
    description: ``,
    fields: [
      {
        id: `title`,
        name: `Title`,
        type: `Symbol`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `boolean`,
        name: `Boolean`,
        type: `Boolean`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `booleanLocalized`,
        name: `Boolean Localized`,
        type: `Boolean`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
      },
    ],
  },
  {
    sys: {
      space: {
        sys: {
          type: `Link`,
          linkType: `Space`,
          id: `k8iqpp6u0ior`,
        },
      },
      id: `date`,
      type: `ContentType`,
      createdAt: `2021-03-01T17:07:02.629Z`,
      updatedAt: `2021-05-20T18:16:28.584Z`,
      environment: {
        sys: {
          id: `master`,
          type: `Link`,
          linkType: `Environment`,
        },
      },
      revision: 2,
    },
    displayField: `title`,
    name: `Date`,
    description: ``,
    fields: [
      {
        id: `title`,
        name: `Title`,
        type: `Symbol`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `date`,
        name: `Date`,
        type: `Date`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `dateTime`,
        name: `Date Time`,
        type: `Date`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `dateTimeTimezone`,
        name: `Date Time Timezone`,
        type: `Date`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `dateLocalized`,
        name: `Date Localized`,
        type: `Date`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
      },
    ],
  },
  {
    sys: {
      space: {
        sys: {
          type: `Link`,
          linkType: `Space`,
          id: `k8iqpp6u0ior`,
        },
      },
      id: `location`,
      type: `ContentType`,
      createdAt: `2021-03-01T17:09:20.579Z`,
      updatedAt: `2021-05-21T10:37:52.205Z`,
      environment: {
        sys: {
          id: `master`,
          type: `Link`,
          linkType: `Environment`,
        },
      },
      revision: 2,
    },
    displayField: `title`,
    name: `Location`,
    description: ``,
    fields: [
      {
        id: `title`,
        name: `Title`,
        type: `Symbol`,
        localized: false,
        required: true,
        disabled: false,
        omitted: false,
      },
      {
        id: `location`,
        name: `Location`,
        type: `Location`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `locationLocalized`,
        name: `Location Localized`,
        type: `Location`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
      },
    ],
  },
  {
    sys: {
      space: {
        sys: {
          type: `Link`,
          linkType: `Space`,
          id: `k8iqpp6u0ior`,
        },
      },
      id: `json`,
      type: `ContentType`,
      createdAt: `2021-03-01T17:09:56.970Z`,
      updatedAt: `2021-05-21T10:36:57.432Z`,
      environment: {
        sys: {
          id: `master`,
          type: `Link`,
          linkType: `Environment`,
        },
      },
      revision: 2,
    },
    displayField: `title`,
    name: `JSON`,
    description: ``,
    fields: [
      {
        id: `title`,
        name: `Title`,
        type: `Symbol`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `json`,
        name: `JSON`,
        type: `Object`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `jsonLocalized`,
        name: `JSON Localized`,
        type: `Object`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
      },
    ],
  },
  {
    sys: {
      space: {
        sys: {
          type: `Link`,
          linkType: `Space`,
          id: `k8iqpp6u0ior`,
        },
      },
      id: `richText`,
      type: `ContentType`,
      createdAt: `2021-03-01T17:11:01.406Z`,
      updatedAt: `2021-11-05T12:56:39.942Z`,
      environment: {
        sys: {
          id: `master`,
          type: `Link`,
          linkType: `Environment`,
        },
      },
      revision: 7,
    },
    displayField: `title`,
    name: `Rich Text`,
    description: ``,
    fields: [
      {
        id: `title`,
        name: `Title`,
        type: `Symbol`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `richText`,
        name: `Rich Text`,
        type: `RichText`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `richTextLocalized`,
        name: `Rich Text Localized`,
        type: `RichText`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `richTextValidated`,
        name: `Rich Text Validated`,
        type: `RichText`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
    ],
  },
  {
    sys: {
      space: {
        sys: {
          type: `Link`,
          linkType: `Space`,
          id: `k8iqpp6u0ior`,
        },
      },
      id: `contentReference`,
      type: `ContentType`,
      createdAt: `2021-03-02T09:17:08.210Z`,
      updatedAt: `2021-05-21T10:36:34.506Z`,
      environment: {
        sys: {
          id: `master`,
          type: `Link`,
          linkType: `Environment`,
        },
      },
      revision: 7,
    },
    displayField: `title`,
    name: `Content Reference`,
    description: ``,
    fields: [
      {
        id: `title`,
        name: `Title`,
        type: `Symbol`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `one`,
        name: `One`,
        type: `Link`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
        linkType: `Entry`,
      },
      {
        id: `oneLocalized`,
        name: `One Localized`,
        type: `Link`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
        linkType: `Entry`,
      },
      {
        id: `many`,
        name: `Many`,
        type: `Array`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
        items: {
          type: `Link`,
          validations: [
            {
              linkContentType: [`contentReference`, `number`, `text`],
            },
          ],
          linkType: `Entry`,
        },
      },
      {
        id: `manyLocalized`,
        name: `Many Localized`,
        type: `Array`,
        localized: true,
        required: false,
        disabled: false,
        omitted: false,
        items: {
          type: `Link`,
          validations: [],
          linkType: `Entry`,
        },
      },
    ],
  },
  {
    sys: {
      space: {
        sys: {
          type: `Link`,
          linkType: `Space`,
          id: `k8iqpp6u0ior`,
        },
      },
      id: `validatedContentReference`,
      type: `ContentType`,
      createdAt: `2021-05-20T15:29:49.734Z`,
      updatedAt: `2021-05-20T15:33:26.620Z`,
      environment: {
        sys: {
          id: `master`,
          type: `Link`,
          linkType: `Environment`,
        },
      },
      revision: 4,
    },
    displayField: `title`,
    name: `ValidatedContentReference`,
    description: ``,
    fields: [
      {
        id: `title`,
        name: `Title`,
        type: `Symbol`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
      },
      {
        id: `oneItemSingleType`,
        name: `One Item: Single Type`,
        type: `Link`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
        linkType: `Entry`,
        validations: [
          {
            linkContentType: [`text`],
          },
        ],
      },
      {
        id: `oneItemManyTypes`,
        name: `One Item: Many Types`,
        type: `Link`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
        linkType: `Entry`,
        validations: [
          {
            linkContentType: [`number`, `text`],
          },
        ],
      },
      {
        id: `oneItemAllTypes`,
        name: `One Item: All Types`,
        type: `Link`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
        linkType: `Entry`,
      },
      {
        id: `multipleItemsSingleType`,
        name: `Multiple Items: Single Type`,
        type: `Array`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
        items: {
          type: `Link`,
          validations: [
            {
              linkContentType: [`text`],
            },
          ],
          linkType: `Entry`,
        },
      },
      {
        id: `multipleItemsManyTypes`,
        name: `Multiple Items: Many Types`,
        type: `Array`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
        items: {
          type: `Link`,
          validations: [
            {
              linkContentType: [`number`, `text`],
            },
          ],
          linkType: `Entry`,
        },
      },
      {
        id: `multipleItemsAllTypes`,
        name: `Multiple Items: All Types`,
        type: `Array`,
        localized: false,
        required: false,
        disabled: false,
        omitted: false,
        items: {
          type: `Link`,
          validations: [],
          linkType: `Entry`,
        },
      },
    ],
  },
]
