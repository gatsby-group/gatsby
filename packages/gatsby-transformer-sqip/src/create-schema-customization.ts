import { GatsbyNode } from "gatsby"
import {
  GraphQLFloat,
  GraphQLInt,
  GraphQLList,
  GraphQLObjectType,
  GraphQLString,
} from "gatsby/graphql"

const SqipMetadataPaletteData = new GraphQLObjectType({
  name: `SqipMetadataPaletteData`,
  fields: {
    population: { type: GraphQLInt },
    rgb: { type: new GraphQLList(GraphQLInt) },
    hsl: { type: new GraphQLList(GraphQLFloat) },
    hex: { type: GraphQLString },
  },
})

export const SqipMetadata = new GraphQLObjectType({
  name: `SqipMetadata`,
  fields: {
    originalWidth: { type: GraphQLInt },
    originalHeight: { type: GraphQLInt },
    palette: {
      type: new GraphQLObjectType({
        name: `SqipMetadataPalette`,
        fields: {
          Vibrant: { type: SqipMetadataPaletteData },
          DarkVibrant: { type: SqipMetadataPaletteData },
          LightVibrant: { type: SqipMetadataPaletteData },
          Muted: { type: SqipMetadataPaletteData },
          DarkMuted: { type: SqipMetadataPaletteData },
          LightMuted: { type: SqipMetadataPaletteData },
        },
      }),
    },
    type: { type: GraphQLString },
    width: { type: GraphQLInt },
    height: { type: GraphQLInt },
  },
})

export const createSchemaCustomization: GatsbyNode["createSchemaCustomization"] =
  ({ actions }) => {
    const { createTypes } = actions
    createTypes([SqipMetadataPaletteData, SqipMetadata])
  }
