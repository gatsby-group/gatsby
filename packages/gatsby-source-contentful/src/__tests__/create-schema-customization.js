// @ts-check
import { createSchemaCustomization } from "../create-schema-customization"
import { contentTypes } from "../__fixtures__/content-types"

const createMockCache = () => {
  return {
    get: jest.fn(key => contentTypes),
  }
}

describe(`create-schema-customization`, () => {
  const actions = { createTypes: jest.fn() }
  const schema = {
    buildObjectType: jest.fn(),
    buildInterfaceType: jest.fn(),
    buildUnionType: jest.fn(),
  }
  const cache = createMockCache()
  const reporter = {
    info: jest.fn(),
    verbose: jest.fn(),
    panic: jest.fn(),
    activityTimer: () => {
      return { start: jest.fn(), end: jest.fn() }
    },
  }

  beforeEach(() => {
    cache.get.mockClear()
    process.env.GATSBY_WORKER_ID = `mocked`
  })

  it(`builds schema based on Contentful Content Model`, async () => {
    await createSchemaCustomization(
      { schema, actions, reporter, cache },
      { spaceId: `testSpaceId` }
    )

    expect(schema.buildObjectType).toHaveBeenCalled()
    expect(schema.buildObjectType.mock.calls).toMatchSnapshot()
    expect(schema.buildInterfaceType).toHaveBeenCalled()
    expect(schema.buildInterfaceType.mock.calls).toMatchSnapshot()
    expect(schema.buildUnionType).toHaveBeenCalled()
    expect(schema.buildUnionType.mock.calls).toMatchSnapshot()
  })
})
