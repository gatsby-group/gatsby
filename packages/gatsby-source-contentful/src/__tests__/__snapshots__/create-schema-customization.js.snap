// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-schema-customization builds schema based on Contentful Content Model 1`] = `
Array [
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "description": Object {
          "type": "String!",
        },
        "displayField": Object {
          "type": "String!",
        },
        "id": Object {
          "type": "ID!",
        },
        "name": Object {
          "type": "String!",
        },
      },
      "interfaces": Array [
        "Node",
      ],
      "name": "ContentfulContentType",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "contentType": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "contentType___NODE",
            },
          },
          "type": "ContentfulContentType",
        },
        "environmentId": Object {
          "type": " String!",
        },
        "firstPublishedAt": Object {
          "type": " Date!",
        },
        "id": Object {
          "type": " String!",
        },
        "locale": Object {
          "type": " String!",
        },
        "publishedAt": Object {
          "type": " Date!",
        },
        "publishedVersion": Object {
          "type": " Int!",
        },
        "spaceId": Object {
          "type": " String!",
        },
        "type": Object {
          "type": " String!",
        },
      },
      "interfaces": Array [
        "Node",
      ],
      "name": "ContentfulSys",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "tags": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "tags___NODE",
            },
          },
          "type": "[ContentfulTag]!",
        },
      },
      "name": "ContentfulMetadata",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "contentful_id": Object {
          "type": "String!",
        },
        "id": Object {
          "type": "ID!",
        },
        "name": Object {
          "type": "String!",
        },
      },
      "interfaces": Array [
        "Node",
      ],
      "name": "ContentfulTag",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "localFile": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
            },
          },
          "type": "File",
        },
      },
      "name": "ContentfulAssetFields",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "contentType": Object {
          "type": "String",
        },
        "description": Object {
          "type": "String",
        },
        "fields": Object {
          "type": "ContentfulAssetFields",
        },
        "fileName": Object {
          "type": "String",
        },
        "height": Object {
          "type": "Int",
        },
        "id": Object {
          "type": "ID!",
        },
        "size": Object {
          "type": "Int",
        },
        "sys": Object {
          "type": "ContentfulSys!",
        },
        "title": Object {
          "type": "String",
        },
        "url": Object {
          "type": "String",
        },
        "width": Object {
          "type": "Int",
        },
      },
      "interfaces": Array [
        "ContentfulReference",
        "Node",
      ],
      "name": "ContentfulAsset",
    },
  ],
  Array [
    Object {
      "fields": Object {
        "block": Object {
          "resolve": [Function],
          "type": "[ContentfulAsset]!",
        },
        "hyperlink": Object {
          "resolve": [Function],
          "type": "[ContentfulAsset]!",
        },
      },
      "name": "ContentfulRichTextAssets",
    },
  ],
  Array [
    Object {
      "fields": Object {
        "block": Object {
          "resolve": [Function],
          "type": "[ContentfulEntry]!",
        },
        "hyperlink": Object {
          "resolve": [Function],
          "type": "[ContentfulEntry]!",
        },
        "inline": Object {
          "resolve": [Function],
          "type": "[ContentfulEntry]!",
        },
      },
      "name": "ContentfulRichTextEntries",
    },
  ],
  Array [
    Object {
      "fields": Object {
        "assets": Object {
          "resolve": [Function],
          "type": "ContentfulRichTextAssets",
        },
        "entries": Object {
          "resolve": [Function],
          "type": "ContentfulRichTextEntries",
        },
      },
      "name": "ContentfulRichTextLinks",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "json": Object {
          "resolve": [Function],
          "type": "JSON",
        },
        "links": Object {
          "resolve": [Function],
          "type": "ContentfulRichTextLinks",
        },
      },
      "name": "ContentfulRichText",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "lat": Object {
          "type": "Float!",
        },
        "lon": Object {
          "type": "Float!",
        },
      },
      "name": "ContentfulLocation",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "raw": "String!",
      },
      "interfaces": Array [
        "Node",
      ],
      "name": "ContentfulText",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "decimal": Object {
          "type": "Float",
        },
        "decimalLocalized": Object {
          "type": "Float",
        },
        "id": Object {
          "type": "ID!",
        },
        "integer": Object {
          "type": "Int",
        },
        "integerLocalized": Object {
          "type": "Int",
        },
        "metadata": Object {
          "type": "ContentfulMetadata!",
        },
        "sys": Object {
          "type": "ContentfulSys!",
        },
        "title": Object {
          "type": "String",
        },
      },
      "interfaces": Array [
        "ContentfulReference",
        "ContentfulEntry",
        "Node",
      ],
      "name": "ContentfulContentTypeNumber",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "id": Object {
          "type": "ID!",
        },
        "longLocalized": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "longLocalized___NODE",
            },
          },
          "type": "ContentfulText",
        },
        "longMarkdown": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "longMarkdown___NODE",
            },
          },
          "type": "ContentfulText",
        },
        "longPlain": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "longPlain___NODE",
            },
          },
          "type": "ContentfulText",
        },
        "metadata": Object {
          "type": "ContentfulMetadata!",
        },
        "short": Object {
          "type": "String",
        },
        "shortList": Object {
          "type": "[String]",
        },
        "shortListLocalized": Object {
          "type": "[String]",
        },
        "shortLocalized": Object {
          "type": "String",
        },
        "sys": Object {
          "type": "ContentfulSys!",
        },
        "title": Object {
          "type": "String",
        },
      },
      "interfaces": Array [
        "ContentfulReference",
        "ContentfulEntry",
        "Node",
      ],
      "name": "ContentfulContentTypeText",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "id": Object {
          "type": "ID!",
        },
        "many": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "many___NODE",
            },
          },
          "type": "[ContentfulAsset]",
        },
        "manyLocalized": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "manyLocalized___NODE",
            },
          },
          "type": "[ContentfulAsset]",
        },
        "metadata": Object {
          "type": "ContentfulMetadata!",
        },
        "one": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "one___NODE",
            },
          },
          "type": "ContentfulAsset",
        },
        "oneLocalized": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "oneLocalized___NODE",
            },
          },
          "type": "ContentfulAsset",
        },
        "sys": Object {
          "type": "ContentfulSys!",
        },
        "title": Object {
          "type": "String",
        },
      },
      "interfaces": Array [
        "ContentfulReference",
        "ContentfulEntry",
        "Node",
      ],
      "name": "ContentfulContentTypeMediaReference",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "boolean": Object {
          "type": "Boolean",
        },
        "booleanLocalized": Object {
          "type": "Boolean",
        },
        "id": Object {
          "type": "ID!",
        },
        "metadata": Object {
          "type": "ContentfulMetadata!",
        },
        "sys": Object {
          "type": "ContentfulSys!",
        },
        "title": Object {
          "type": "String",
        },
      },
      "interfaces": Array [
        "ContentfulReference",
        "ContentfulEntry",
        "Node",
      ],
      "name": "ContentfulContentTypeBoolean",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "date": Object {
          "extensions": Object {
            "dateformat": Object {},
          },
          "type": "Date",
        },
        "dateLocalized": Object {
          "extensions": Object {
            "dateformat": Object {},
          },
          "type": "Date",
        },
        "dateTime": Object {
          "extensions": Object {
            "dateformat": Object {},
          },
          "type": "Date",
        },
        "dateTimeTimezone": Object {
          "extensions": Object {
            "dateformat": Object {},
          },
          "type": "Date",
        },
        "id": Object {
          "type": "ID!",
        },
        "metadata": Object {
          "type": "ContentfulMetadata!",
        },
        "sys": Object {
          "type": "ContentfulSys!",
        },
        "title": Object {
          "type": "String",
        },
      },
      "interfaces": Array [
        "ContentfulReference",
        "ContentfulEntry",
        "Node",
      ],
      "name": "ContentfulContentTypeDate",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "id": Object {
          "type": "ID!",
        },
        "location": Object {
          "type": "ContentfulLocation",
        },
        "locationLocalized": Object {
          "type": "ContentfulLocation",
        },
        "metadata": Object {
          "type": "ContentfulMetadata!",
        },
        "sys": Object {
          "type": "ContentfulSys!",
        },
        "title": Object {
          "type": "String",
        },
      },
      "interfaces": Array [
        "ContentfulReference",
        "ContentfulEntry",
        "Node",
      ],
      "name": "ContentfulContentTypeLocation",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "id": Object {
          "type": "ID!",
        },
        "json": Object {
          "type": "JSON",
        },
        "jsonLocalized": Object {
          "type": "JSON",
        },
        "metadata": Object {
          "type": "ContentfulMetadata!",
        },
        "sys": Object {
          "type": "ContentfulSys!",
        },
        "title": Object {
          "type": "String",
        },
      },
      "interfaces": Array [
        "ContentfulReference",
        "ContentfulEntry",
        "Node",
      ],
      "name": "ContentfulContentTypeJson",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "id": Object {
          "type": "ID!",
        },
        "metadata": Object {
          "type": "ContentfulMetadata!",
        },
        "richText": Object {
          "type": "ContentfulRichText",
        },
        "richTextLocalized": Object {
          "type": "ContentfulRichText",
        },
        "richTextValidated": Object {
          "type": "ContentfulRichText",
        },
        "sys": Object {
          "type": "ContentfulSys!",
        },
        "title": Object {
          "type": "String",
        },
      },
      "interfaces": Array [
        "ContentfulReference",
        "ContentfulEntry",
        "Node",
      ],
      "name": "ContentfulContentTypeRichText",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "id": Object {
          "type": "ID!",
        },
        "many": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "many___NODE",
            },
          },
          "type": "[UnionContentfulContentReferenceNumberText]",
        },
        "manyLocalized": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "manyLocalized___NODE",
            },
          },
          "type": "[ContentfulEntry]",
        },
        "metadata": Object {
          "type": "ContentfulMetadata!",
        },
        "one": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "one___NODE",
            },
          },
          "type": "ContentfulEntry",
        },
        "oneLocalized": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "oneLocalized___NODE",
            },
          },
          "type": "ContentfulEntry",
        },
        "sys": Object {
          "type": "ContentfulSys!",
        },
        "title": Object {
          "type": "String",
        },
      },
      "interfaces": Array [
        "ContentfulReference",
        "ContentfulEntry",
        "Node",
      ],
      "name": "ContentfulContentTypeContentReference",
    },
  ],
  Array [
    Object {
      "extensions": Object {
        "dontInfer": Object {},
      },
      "fields": Object {
        "id": Object {
          "type": "ID!",
        },
        "metadata": Object {
          "type": "ContentfulMetadata!",
        },
        "multipleItemsAllTypes": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "multipleItemsAllTypes___NODE",
            },
          },
          "type": "[ContentfulEntry]",
        },
        "multipleItemsManyTypes": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "multipleItemsManyTypes___NODE",
            },
          },
          "type": "[UnionContentfulNumberText]",
        },
        "multipleItemsSingleType": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "multipleItemsSingleType___NODE",
            },
          },
          "type": "[ContentfulContentTypeText]",
        },
        "oneItemAllTypes": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "oneItemAllTypes___NODE",
            },
          },
          "type": "ContentfulEntry",
        },
        "oneItemManyTypes": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "oneItemManyTypes___NODE",
            },
          },
          "type": "UnionContentfulNumberText",
        },
        "oneItemSingleType": Object {
          "extensions": Object {
            "link": Object {
              "by": "id",
              "from": "oneItemSingleType___NODE",
            },
          },
          "type": "ContentfulContentTypeText",
        },
        "sys": Object {
          "type": "ContentfulSys!",
        },
        "title": Object {
          "type": "String",
        },
      },
      "interfaces": Array [
        "ContentfulReference",
        "ContentfulEntry",
        "Node",
      ],
      "name": "ContentfulContentTypeValidatedContentReference",
    },
  ],
]
`;

exports[`create-schema-customization builds schema based on Contentful Content Model 2`] = `
Array [
  Array [
    Object {
      "fields": Object {
        "id": Object {
          "type": "ID!",
        },
        "sys": Object {
          "type": "ContentfulSys!",
        },
      },
      "interfaces": Array [
        "Node",
      ],
      "name": "ContentfulReference",
    },
  ],
  Array [
    Object {
      "fields": Object {
        "id": Object {
          "type": "ID!",
        },
        "metadata": Object {
          "type": "ContentfulMetadata!",
        },
        "sys": Object {
          "type": "ContentfulSys!",
        },
      },
      "interfaces": Array [
        "ContentfulReference",
        "Node",
      ],
      "name": "ContentfulEntry",
    },
  ],
]
`;

exports[`create-schema-customization builds schema based on Contentful Content Model 3`] = `
Array [
  Array [
    Object {
      "name": "UnionContentfulContentReferenceNumberText",
      "types": Array [
        "ContentfulContentTypeContentReference",
        "ContentfulContentTypeNumber",
        "ContentfulContentTypeText",
      ],
    },
  ],
  Array [
    Object {
      "name": "UnionContentfulNumberText",
      "types": Array [
        "ContentfulContentTypeNumber",
        "ContentfulContentTypeText",
      ],
    },
  ],
]
`;
